src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,669 @@
 #!/usr/bin/env python2.7
-
-""" 
-GATK HaplotypeCaller Genotyping Mode
-
-            Tree Structure of GATK Pipeline
-            0 --> 1 --> 2 --> 3 --> 4
-                                    |
-                                    5
-                                   / \
-                                  6   7
-                                 /     \
-                                8       9
-0 = Start Node
-1 = Download Reference
-2 = Index Reference
-3 = Reference Dictionary
-4 = Index Samples
-5 = HaplotypeCaller SNP & Indel
-6 = VariantRecalibrator SNPs
-7 = VariantRecalibrator Indels
-8 = ApplyRecalibration SNPs
-9 = ApplyRecalibration Indels
-
-===================================================================
-:Dependencies:
-curl            - apt-get install curl
-docker          - apt-get install docker (or 'docker.io' for linux)
-toil            - pip install --pre toil
-"""
 from __future__ import print_function
-
 import argparse
-import errno
+from copy import deepcopy
+import csv
 import multiprocessing
 import os
-import shutil
-import subprocess
+import re
 import sys
-import textwrap
-from collections import OrderedDict
+from urlparse import urlparse
 
+from bd2k.util.humanize import human2bytes
+from bd2k.util.processes import which
+import synapseclient
+from toil.job import Job, PromisedRequirement
 import yaml
-from toil.job import Job
-from toil_scripts import download_from_s3_url
+
+from toil_scripts.gatk_germline.germline_config import generate_config, generate_manifest
+from toil_scripts.gatk_germline.hard_filter import hard_filter_pipeline
+from toil_scripts.gatk_germline.vqsr import vqsr_pipeline
 from toil_scripts.lib import require
+from toil_scripts.lib.files import get_files_from_filestore, upload_or_move_job
 from toil_scripts.lib.programs import docker_call
-from toil_scripts.lib.urls import s3am_upload
+from toil_scripts.lib.urls import download_url_job
 from toil_scripts.rnaseq_cgl.rnaseq_cgl_pipeline import generate_file
+from toil_scripts.tools.aligners import run_bwakit
+from toil_scripts.tools.indexing import run_samtools_faidx
+from toil_scripts.tools.preprocessing import run_germline_preprocessing, run_samtools_sort, \
+    run_samtools_index, run_samtools_view, run_picard_create_sequence_dictionary
+from toil_scripts.tools.variant_annotation import run_oncotator
+
+
+def download_and_run_sample(job, uuid, url, config, url2=None, rg_line=None):
+    """
+    Downloads reference files and runs the GATK best practices germline pipeline for one sample.
+    See the germline config file for necessary parameters.
+
+    :param JobFunctionWrappingJob job: Toil Job instance
```

Comment:
Might add that this is passed to every job automatically (other pipelines use this comment at Hannes's suggestion)